onnx/model_container.py

```diff
@@ -0,0 +1,324 @@
+                    f.write(np_tensor.tobytes())
+
+        with open(file_path, "wb") as f:
+            f.write(copy.SerializeToString())
+        return copy
+
+    def save(
+        self,
+        file_path: str,
+        all_tensors_to_one_file: bool = False,
+    ) -> onnx.ModelProto:
+        """
+        Save the large model.
+        The function returns a ModelProto,
+        the current one if the model did not need any modification,
+        a modified copy of it if it required changes such as giving file names
+        to every external tensor.
+
+        Arguments:
+            file_path: model file
+            all_tensors_to_one_file: saves all large tensors in one file or
+                one file per lerge tensor
+
+        Returns:
+            the saved ModelProto
+        """
+        return self._save_external(
+            file_path, all_tensors_to_one_file=all_tensors_to_one_file
+        )
+
+    def load(self, file_path: str, load_large_initializers: bool = True):
+        """
+        Load the large model.
+
+        Arguments:
+            file_path: model file
+            load_large_initializers: loads the large initializers,
+                if not done, the model is incomplete but it can be used to
+                look into the model without executing it and method
+                :meth:`_load_large_initializers` can be used to load them later
+        """
+        self.model_proto_ = onnx.load_model(file_path, load_external_data=False)
+        if load_large_initializers:
+            self._load_large_initializers(file_path)
+
+    def _load_large_initializers(self, file_path):
+        """
+        Loads large initializers.
+
+        Arguments:
+            file_path: model file, the weight are expected to be in the same folder as this file
+        """
+        if self.model_proto_ is None:
+            raise RuntimeError("A model must be loaded before loading the weights.")
+        self.large_initializers = {}
+        base_dir = os.path.dirname(file_path)
+        for i, tensor in enumerate(ext_data._get_all_tensors(self.model_proto_)):
+            if not ext_data.uses_external_data(tensor):
+                continue
+
+            info = ext_data.ExternalDataInfo(tensor)
+            file_location = ext_data._sanitize_path(info.location)
+            external_data_file_path = os.path.join(base_dir, file_location)
+            key = f"#t{i}"
+            _set_external_data(tensor, location=key)
+
+            with open(external_data_file_path, "rb") as data_file:
+                if info.offset:
+                    data_file.seek(info.offset)
+
+                raw_data = (
+                    data_file.read(info.length) if info.length else data_file.read()
+                )
+
+                dtype = onnx.helper.tensor_dtype_to_np_dtype(tensor.data_type)
+                shape = tuple(tensor.dims)
+                self.large_initializers[key] = np.frombuffer(
```

Comment:
We may need to do a byteswap for big-endian machines here?

Reply:
done